src/main/java/de/internetsicherheit/brl/bloxberg/cache/gui/BlockDataExtractor.java

```diff
@@ -59,9 +59,9 @@ public void writeOutTransactions(int blockNumber, SequenceWriter seqWriter) thro
         EthBlock.Block ethBlock = client.getEthBlock(blockBigInteger).getBlock();
 
         List transList = ethBlock.getTransactions();
-        ListIterator it = transList.listIterator();
+        ListIterator<Transaction> it = transList.listIterator();
         while (it.hasNext()) {
-            seqWriter.write(new ReducedTransObject((Transaction) it.next()));
+            seqWriter.write(new ReducedTransObject(it.next(), ethBlock));
```

Comment:
Can we replaced by 
```java
        List<ReducedTransObject> convertedObjects = ethBlock.getTransactions()
                .stream()
                .map(it -> new ReducedTransObject((EthBlock.TransactionObject) it, ethBlock))
                .collect(Collectors.toList());
        
        seqWriter.writeAll(convertedObjects);
```

Reply:
yes. so i think streams /lambda expressions are kinda easy to read but i have a hard time writing them myself.
also, is type
_EthBlock.TransactionObject_
equivalent to type
_Transaction_
?

Reply:
```
public static class TransactionObject extends Transaction
            implements TransactionResult<Transaction>
```
So it is probably the concrete implementing type, at least that's what I saw when I debugged this code point.

If stream expressions are hard to write for you, it just shows that you need more practice in writing them :slightly_smiling_face: